coalib/output/ConsoleInteraction.py

```diff
@@ -218,19 +221,54 @@ def print_result(console_printer,
                          "not a valid derivative of the coala result "
                          "class.")
         return
+    if section is None:
+        raise ValueError("Section has to be intializied.")
 
     console_printer.print(format_lines("[{sev}] {bear}:".format(
         sev=RESULT_SEVERITY.__str__(result.severity), bear=result.origin)),
         color=RESULT_SEVERITY_COLORS[result.severity])
     console_printer.print(format_lines(result.message), delimiter="\n")
 
     if interactive:
+        cli_actions = CLI_ACTIONS
+        show_patch_action = ShowPatchAction()
+        if show_patch_action.is_applicable(result, file_dict, file_diff_dict):
+            diff_size = sum(len(diff) for diff in result.diffs.values())
+            if diff_size <= DIFF_EXCERPT_MAX_SIZE:
+                show_patch_action.apply_from_section(result,
+                                                     file_dict,
+                                                     file_diff_dict,
+                                                     section)
+                cli_actions = tuple(action for action in cli_actions
+                                    if not isinstance(action, ShowPatchAction))
+            else:
+                print_diffs_info(result.diffs, file_dict, file_diff_dict,
+                                 console_printer)
         acquire_actions_and_apply(console_printer,
                                   log_printer,
                                   section,
                                   file_diff_dict,
                                   result,
-                                  file_dict)
+                                  file_dict,
+                                  cli_actions)
+
+
+def print_diffs_info(diffs, original_file_dict, file_diff_dict, printer):
+    for filename, diff in sorted(diffs.items()):
+        if filename in file_diff_dict:
+            current_file = file_diff_dict[filename].modified
+        else:
+            current_file = original_file_dict[filename]
+        new_file = (diff + file_diff_dict.get(filename, Diff([]))).modified
+        unified_diff = list(difflib.unified_diff(current_file, new_file))[2:]
```

Comment:
actually, rather than creating a unified diff and all that it'd be way better and more useful to add a diffstat function to the Diff class and just sum up the added lines, deleted lines and for a change add 1 to both values. No need to fiddle around with file dicts :)